test/OpenTelemetry.Extensions.Enrichment.Tests/EnrichmentExtensionsTests.cs

```diff
@@ -0,0 +1,164 @@
+// <copyright file="EnrichmentExtensionsTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Extensions.Enrichment.Tests;
+
+public sealed class EnrichmentExtensionsTests
+{
+    private const string SourceName = nameof(EnrichmentExtensionsTests);
+
+    [Fact]
+    public void TracerProviderBuilder_AddTraceEnricherT_RegistersEnricher()
+    {
+        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+            .AddSource(SourceName)
+            .AddTraceEnricher<MyTraceEnricher>()
+            .AddTraceEnricher<MyTraceEnricher2>()
+            .Build();
+
+        using var source = new ActivitySource(SourceName);
+        using (var activity = source.StartActivity(SourceName))
+        {
+            activity.Stop();
+            Assert.Equal(1, (int)activity.GetTagItem(MyTraceEnricher.Key));
```

Comment:
I dont think this test is adding any value on top of the tests using inmemoryexporter. Probably best to get rid of this one infavor of InMemoryExporter ones.

Reply:
This test method is testing AddTracerEnricher<T>(), other test methods are testing different extension methods such as AddTraceEnricher() and so on. 

Anyway, updated all tests to use InMemoryExporter instead of querying the Activity object directly.